### ISSUE TITLE
Add LiquibaseDatabase to support interpreting the liquibase XML format in the code generator

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#2286]
Flyway migrations can be easily simulated by jOOQ's `DDLDatabase`, and with jOOQ 3.12 now also with `DSLContext.meta(Source...)`, in case of which DDL scripts in text form are parsed and interpreted / simulated either with an H2 in-memory database, or with our new DDL interpreter (starting from jOOQ 3.13).

Liquibase migrations are usually expressed in XML form. A sample set of changes can be seen here:
https://github.com/khartec/waltz/tree/master/waltz-data/src/main/ddl/liquibase

More information can be found here:
http://www.liquibase.org/manual/home

The feature request is here:
https://groups.google.com/d/topic/jooq-user/CQv7i4jDyCc/discussion

**Comment by knutwannheden** [comment]
A project using Liquibase: https://github.com/khartec/waltz/tree/master/waltz-data/src/main/ddl/liquibase

**Comment by lukaseder** [comment]
@knutwannheden See my issue description ;-)

**Comment by lukaseder** [comment]
Rather than interpreting the XML file on our own, we could (for now) add an actual Liquibase dependency to `jOOQ-meta-extensions`. It appears that we can run the Liquibase migration against an in-memory H2 database very easily:

```java
Database database = DatabaseFactory.getInstance()
    .findCorrectDatabaseImplementation( 
        new JdbcConnection( connection ) 
    );
Liquibase liquibase = new Liquibase( 
    "db.changelog.xml", 
    new FileSystemResourceAccessor(), 
    database 
);
liquibase.update( "" );
```

Source: https://stackoverflow.com/q/26139744/521799

In the future, we will want to improve this, just like with the `DDLDatabase`, by intercepting the actual SQL statements in the target dialect, and interpreting them with the DDL interpreter (see #8528).

But offering a quick win already now will add substantial value today, and work as reliably as the existing H2 based `DDLDatabase`.

**Comment by lukaseder** [comment]
This seems to work effortlessly! We probably want to exclude the `DATABASECHANGELOG` and `DATABASECHANGELOGLOCK` tables generated by Liquibase, by default. If users want these to be generated, they can set a `includeLiquibaseTables=true` property.

**Comment by lukaseder** [comment]
Implemented this for jOOQ 3.13. This issue only covers Liquibase support as a source for code generation meta information. Runtime meta information derived from Liquibase XML files will be supported as well, through #9360
